codice/ddf
# DDF-4570 Persist updated Solr password

#### What does this PR do?
- Master port of 2.13.x PR: #4633. However, it includes design changes not present in PR #4663.
- To prevent he admin module from depending directly in the Solr password update module, an interface was introduced.
- The SystemPropertiesAdminInterceptor interface was created as an OSGI service interface. The SystemPropertiesAdminMBean implementation will give all SystemPropertiesAdminInterceptor service providers a chance to update system properties before saving the properties to disk.
- The Solr password update modules uses blueprint to register a SystemPropertiesAdminInterceptor service provider.

#### Who is reviewing it? 
@austinsteffes @tyler30clemens @brjeter @pklinef 
#### Select relevant component teams: 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@brendan-hofmann
@brjeter
@clockard


#### How should this be tested?

1. Build DDF.
2. Go through the installation process and let DDF restart it self.

Expected behavior: DDF can communicate with Solr after the restart.
Post-condition: The property `solr.password` in file `etc/custom.system.properties` begins with the literal `ENC(`

Example:
```
aaron@aaron-dell:/sandbox/ddf-2.13.8-SNAPSHOT$ grep "solr.password" etc/custom.system.properties 
solr.password=admin
aaron@aaron-dell:/sandbox/ddf-2.13.8-SNAPSHOT$ grep "solr.password" etc/custom.system.properties 
solr.password = ENC(AWADVsbtqoPKVRVcDK5CsW2vSyvVTWbe4AzFQjbBycYx8KwEXk3KKyTQ7nONwWubqNFXyDniyUMkE9+05EZN3JM=)
```

For GH Issues:
Fixes: #4570

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
